src/reference/antora/modules/ROOT/pages/cloudevents.adoc

```diff
@@ -0,0 +1,268 @@
+Then it will search for any header that contains a key of 'span-id' as well as 'user-id' and add those headers as extensions if found.
+
+[source,java]
+----
+String[] extensionPatterns = {"trace*", "span-id", "user-id"};
+return new ToCloudEventTransformer(extensionPatterns);
+----
+
+[[cloudevent-attribute-defaults]]
+====== Default Values
+The following table contains the Attribute names and the value returned by the default `Expression`s.
+
+|===
+| Attribute Name | Default Value
+
+| `id`
+| the id of the message.
+
+| `source`
+| Prefix of "/spring/" followed by the appName a period then the name of the transformer's bean. i.e. `/spring/myapp.toCloudEventsTransformerBean`
+
+| `type`
+| "spring.message"
+
+| `dataContentType`
+| The `contentType` of the message.
+
+| `dataSchema`
+| `null`
+
+| `subject`
+| `null`
+
+| `time`
+| The time the CloudEvent message is created
+|===
+
+[[cloudevent-transformer-integration]]
+==== Integration with Spring Integration Flows
+
+The CloudEvent transformer integrates with Spring Integration flows:
+
+====== Basic Flow
+
+[source,java]
+----
+@Bean
+public IntegrationFlow cloudEventTransformFlow() {
+    return IntegrationFlows
+        .from("inputChannel")
+        .transform(cloudEventTransformer())
+        .channel("outputChannel")
+        .get();
+}
+----
+
+[[cloudevent-transformer-transformation-process]]
+===== Transformation Process
+
+The transformer follows the process below:
+
+1. **CloudEvent Building**: Build CloudEvent attributes
+2. **Extension Extraction**: Build the CloudEvent extensions using the array of extensionExpressions passed into the constructor.
+3. **Format Conversion**: Apply the specified `EventFormat` based on the message's `contentType to create the CloudEvent.
+
+[[cloudevent-transformer-examples]]
+===== Examples
+
+[[cloudevent-transformer-example-basic]]
+====== Basic Message Transformation
+
+[source,java]
+----
+// Input message with headers
+Message<String> inputMessage = MessageBuilder
+    .withPayload("Hello CloudEvents")
+    .withHeader("contentType", "application/octet-stream")
+    .build();
+// Transformer with extension patterns
+ToCloudEventTransformer transformer = new ToCloudEventTransformer();
+
+// Transform to CloudEvent
+Object cloudEventMessage = transformer.transform(inputMessage);
+----
+
+[[eventformats]]
+===== EventFormats
+
+The `ToCloudEventsTransformer` uses `EventFormat` s to serialize the CloudEvent into the message's payload.
+The `EventFormat` s used by the `ToCloudEventsTransformer` are obtained from the classpath of the project.
+The `EventFormat` s that are available from the https://github.com/cloudevents/sdk-java[CloudEvents SDK] project are as follows:
+
+[[jsonformat]]
+====== JsonFormat
+The following dependency can be used to include this `JsonFormat` in your project.
```

Comment:
No sticking to the title, please.
And that is, of course, if we move on with "may sub-titles" approach 😄